### ISSUE TITLE
Cannot create new project on CLI 12.1.0-next.3 / dependency issues

### ISSUE THREAD

**Issue by pwilkowski** [angular/angular-cli#21028]
# 🐞 Bug report

### Command (mark with an `x`)

- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

 Yes, the previous version in which this bug was not present was: 11

### Description

trying to create new project using `ng new` then selecting YES for angular routing and SCSS results in error:

`ERESOLVE unable to resolve dependency tree `

## 🔬 Minimal Reproduction

`ng new`
Yes for angular routing
SCSS for styles

## 🔥 Exception or Error

<pre><code>
⠏ Installing packages (npm)...npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree 
npm ERR!
npm ERR! While resolving: ng12@0.0.0
npm ERR! Found: @angular/compiler-cli@12.1.0-next.4
npm ERR! node_modules/@angular/compiler-cli
npm ERR!   dev @angular/compiler-cli@"~12.1.0-next.3" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/compiler-cli@"^12.0.0-next" from @angular-devkit/build-angular@12.1.0-next.4
npm ERR! node_modules/@angular-devkit/build-angular
npm ERR!   dev @angular-devkit/build-angular@"~12.1.0-next.3" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!    \AppData\Local\npm-cache\_logs\2021-06-03T01_01_44_281Z-debug.log
✖ Package install failed, see above.
The Schematic workflow failed. See above.
</code></pre>

## 🌍 Your Environment

<pre><code>
     _                      _                 ____ _     ___ 
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | | 
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | | 
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 12.1.0-next.3
Node: 14.16.0
Package Manager: npm 7.15.1
OS: win32 x64

Angular: 
... 

Package                      Version
------------------------------------------------------ 
@angular-devkit/architect    0.1201.0-next.3 (cli-only)
@angular-devkit/core         12.1.0-next.3 (cli-only)  
@angular-devkit/schematics   12.1.0-next.3 (cli-only)  
@schematics/angular          12.1.0-next.3 (cli-only) 
</code></pre>

**Anything else relevant?**

Before installing 12.1.0-next.3 I was getting TONS of deprecate warnings with angular CLI 11, tried to somehow resolve them by removing npm/yarn cache, fiddling around with angular version, reinstalling angular CLI, none of that helped. Then tried CLI 12 however it still had deprecation warnings.

Normally I wouldn't be bothered by couple of those however after seeing 40+ I realized something is clearly wrong. Unfortunately next did not help either, now I cannot even create new project. I was not sure whatever I should create issue for this or not. But this is how existing project that has previously worked (angular version 11.2.14), looks after installing dependencies using yarn:

```
warning @angular-devkit/build-angular > webpack-dev-server > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning @angular-devkit/build-angular > webpack-dev-server > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
warning @angular-devkit/build-angular > stylus > css-parse > css > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning @angular-devkit/build-angular > stylus > css-parse > css > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning @angular-devkit/build-angular > webpack-dev-middleware > webpack-log > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  Se
e https://v8.dev/blog/math-random for details.
warning @angular-devkit/build-angular > webpack-dev-server > sockjs > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https:/
/v8.dev/blog/math-random for details.
warning @angular-devkit/build-angular > webpack > watchpack > watchpack-chokidar2 > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
warning @angular-devkit/build-angular > stylus > css-parse > css > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
warning @angular/cli > universal-analytics > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random
for details.
warning @angular/cli > universal-analytics > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @angular/cli > universal-analytics > request > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/ma
th-random for details.
warning @angular/cli > universal-analytics > request > har-validator@5.1.5: this library is no longer supported
warning @angular/cli > pacote > @npmcli/run-script > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning @types/bootstrap > popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
[2/4] Fetching packages...
info fsevents@2.3.2: The platform "win32" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "win32" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > bootstrap@4.6.0" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.6.0" has unmet peer dependency "popper.js@^1.16.1".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default@5.1.2" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > cssnano-utils@2.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-colormin@5.2.0" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-convert-values@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-discard-comments@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-discard-duplicates@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-merge-longhand@5.0.2" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-discard-overridden@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-discard-empty@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-merge-rules@5.0.2" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-minify-gradients@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-minify-params@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-minify-selectors@5.1.0" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-charset@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-display-values@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-string@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-repeat-style@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-positions@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-timing-functions@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-unicode@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-url@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-normalize-whitespace@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-ordered-values@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-reduce-initial@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-reduce-transforms@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-unique-selectors@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-svgo@5.0.2" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-minify-font-values@5.0.1" has incorrect peer dependency "postcss@^8.2.15".
warning "@angular-devkit/build-angular > cssnano > cssnano-preset-default > postcss-merge-longhand > stylehacks@5.0.1" has incorrect peer dependency "postcss@^8.2.15".

```

**Comment by alan-agius4** [comment]
Hi @pwilkowskil,

Can you run `npm ls postcss`, I am suspecting NPM 7 is hoisting the packages incorrectly. NPM 7 has a number of defects which causes incorrect hoisting.

**Comment by alan-agius4** [comment]
I deleted my previous comment, cause I got slightly confused with the report. It seems that the are 2 different issues being reported here. 

V12 with NPM 7
This is an oversight that we didn't update the peer dependencies to cater for 12.1.0 prerelease version.

V11 with Yarn
This is related to https://github.com/angular/angular-cli/issues/20888 and a fix will be available in `@angular-devkit/build-angular` version `0.1102.14`.

**Comment by pwilkowski** [comment]
@alan-agius4 

Technically the section of "Anything else relevant?" is how I got into this issue in the first place. I am not hellbent on keeping angular on version 11.

Updating NPM to version 7 was just **one of the things** I have tried. Technically the dependencies issues are also present on NPM (Before updating to 7 that broke it completely).

To share some more light on this, 
following issues are **ONLY about 12.1.0-next.3**:

1. After `ng new` failed i tried to simply use `yarn` inside of project
2. `npm i` still fails with exact same message
3. Two warnings disappear when i put "resolution":
  ```
  "resolutions": {
		  "chokidar": "^3.5.0"
  }
  ```
  
  ```
  warning @angular-devkit/build-angular > webpack-dev-server > chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
  warning @angular-devkit/build-angular > webpack-dev-server > chokidar > fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.```
  ```
  So I will probably open new issue (or find existing one) in `webpack-dev-server` since I have confirmed that this package alone causes at least couple warnings.

4. `pacote` requires `@npmcli/run-script` to be updated so it is not angular issue
5. `stylus` is suppose to be gone? Am I missing something? https://twitter.com/Splaktar/status/1387646440026087425 Or is it simply not pushed/released yet?

```
warning @angular-devkit/build-angular > stylus > css-parse > css > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning @angular-devkit/build-angular > stylus > css-parse > css > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
warning @angular-devkit/build-angular > stylus > css-parse > css > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated```

If you need any more information, let me know.

**Comment by alan-agius4** [comment]
That tweet does seem to send an incorrect message. //cc @Splaktar.`stylus` support is not removed, it is deprecated in v12. What was removed was the possibility to use `stylus` in new applications.

Regarding the warnings, indeed these seem to be upstream related and cannot be addressed directly by us.

**Comment by Splaktar** [comment]
Thank you for the clarification. Clarifying tweet posted: https://twitter.com/Splaktar/status/1400522600003117057

**Comment by crfrolik** [comment]
> This is related to #20888 and a fix will be available in `@angular-devkit/build-angular` version `0.1102.14`.

Sounds great, but this can't happen fast enough - is there an ETA on 0.1102.14?

**Comment by alan-agius4** [comment]
@dgp1130 will be cutting an LTS release for version 11 later on today.

**Comment by dgp1130** [comment]
@crfrolik, [CLI v11.2.14](https://github.com/angular/angular-cli/releases/tag/v11.2.14) was just released. We had some CI issues this week which slowed things down a bit, but @clydin was able to figure those out to unblock the release.
